clients/go/admin/pkce/token_cache.go

```diff
@@ -0,0 +1,14 @@
+package pkce
+
+import "golang.org/x/oauth2"
+
+//go:generate mockery -all -case=underscore
+
+// TokenCache defines the interface needed to cache and retrieve oauth tokens.
```

Comment:
why is this even an interface? ease of testing?

Reply:
We actually want another implementation (keyring based like the one you added to flyte-cli) but didn't want to put it here with the dependencies it brings on... so left it in flytectl...

Reply:
thanks for explaining!